crates/store/src/db/mod.rs

```diff
@@ -224,6 +224,24 @@ impl Db {
             })?
     }
 
+    /// Loads multiple block headers from the DB.
+    #[instrument(target = "miden-store", skip_all, ret(level = "debug"), err)]
+    pub async fn select_many_block_headers(
```

Comment:
Or maybe `multiple_` instead? I also considered overloading the `select_block_headers` to take an optional vec instead.